### DISCUSSION TITLE
About supported file format

### DISCUSSION THREAD

**Discussion by momo91377** [web-infra-dev/rspress#1486] [Q&A]
I want to insert an offline excel sheet inside a document package written using rspress, can it be referenced on the specific document page, using the jump link writing method won't work because the suffix will be added to the html

**Comment by Timeless0911** [comment]
https://rspress.dev/api/config/config-basic#routecleanurls  You can use this config to clean urls.

**Reply by momo91377** [reply]
Can it be set for one place, but other parts need to keep this setting, we can't use the global setting.

**Reply by Timeless0911** [reply]
No, I think you can use some redirect behaviour to an url.

**Reply by momo91377** [reply]
I only have a jump link to insert offline Excel for one of the html pages, and this file will be put into the package, can't it be done at the moment?

**Reply by Timeless0911** [reply]
You can extend asset types via `builderConfig` in your `rspress.config`

https://rsbuild.dev/guide/basic/static-assets#extend-asset-types

**Reply by momo91377** [reply]
Got it, we have a try, thanks

**Reply by momo91377** [reply]
Does this configuration also work in rspress?

**Reply by Timeless0911** [reply]
You mean you use `[foo](/foo.xlsx)` in mdx file, and the jump link will be added a `.html` suffix, right?

If so, this config is not fit with your scenrio. You have three ways to go:

1. Set `route.cleanUrls=true`, which is a modern way of website
2. Use an external url of your excel file, you can also put it in your website project, but you should upload static resources to cdn, and use an external url which starts with `http(s)://`.
3. Write a custom `FileLink` component to do it yourself.

**Reply by momo91377** [reply]
是这样的，我们整体内容是使用rspress  mdx的格式写的，现在需求是想要在生成的页面中插入一个excel的链接，然后希望是能把它放在包内，但是按照正常的路径引用，是不支持Excel格式，所以是想咨询这个有没有什么办法可以解决

**Reply by Timeless0911** [reply]
上面不是给了三个解决方案吗

**Reply by Timeless0911** [reply]
https://rspress.dev/guide/basic/static-assets#other-static-assets 不管是什么格式的静态资源，处理方式都差不多，你要么放在 server 里要么放在 cdn 上，然后再去调整对应的构建方式与配置，如果内置的链接语法不支持，你可以自己写一个对应的组件去插入这个链接
